geth/signal/signals_test.go

```diff
@@ -0,0 +1,50 @@
+package signal
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestNodeCrashEventJSONMarshalling(t *testing.T) {
+	errorMsg := "TestNodeCrashEventJSONMarshallingError"
+	expectedJSON := fmt.Sprintf(`{"error":"%s"}`, errorMsg)
+	nodeCrashEvent := &NodeCrashEvent{
+		Error: errors.New(errorMsg),
+	}
+	marshalled, err := json.Marshal(nodeCrashEvent)
+	require.NoError(t, err)
+	require.Equal(t, expectedJSON, string(marshalled))
+}
+
+func TestEnvelopeJSONMarshalling(t *testing.T) {
+	errorMsg := "TestEnvelopeJSONMarshallingError"
+	expectedJSON := fmt.Sprintf(`{"event":{"error":"%s"},"type":"node.crashed"}`, errorMsg)
+	envelope := &Envelope{
+		Type: EventNodeCrashed,
+		Event: NodeCrashEvent{
```

Comment:
this should be a pointer

Reply:
You're right. `NodeCrashEvent` is never defined as a pointer in the other assignments in this repo, so I'm updating `MarshalJSON` to make sure the newly defined struct uses a pointer for consistency with the original `interface{}` field.

Reply:
Why does it need to be a pointer? I'm updating the PR so that the third test (`TestEnvelopeJSONMarshallingWithoutNodeCrashEvent`) also doesn't use a pointer for the struct assigned to the 'Event' field for consistency.

Are you suggesting that I modify MarshalJSON as such?
```
		return json.Marshal(&struct {
			*Alias
			Event *NodeCrashEvent `json:"event"`
		}{
			Alias: (*Alias)(e),
			Event: &event,
		})
```

Reply:
Correct. Removing `(*Envelope) MarshalJSON` and modifying `(*NodeCrashEvent) MarshalJSON` to be defined by value. So no change here.